### PR TITLE
PR request

### PR DESCRIPTION
**Pull request by riyazah18**
request method name changed to processRequest and logging enhancement.

### CODE REVIEW

File: src/main/java/com/veryfi/VeryfiClient.java [hunk 1]
```diff
@@ -221,8 +225,9 @@ public String getDocuments() throws VeryfiClientException {
      * @throws VeryfiClientException if there is any error in making request to veryfi APIs
      */
     public String getDocument(String documentId) throws VeryfiClientException {
+        logger.info("Getting Document with documentId["+documentId+"]");
```
Comment (AsadShakeel) [inline src/main/java/com/veryfi/VeryfiClient.java]: We can use the logging formatting here like
`logger.info("Getting Document with documentId[{}]", documentId);`

File: src/main/java/com/veryfi/VeryfiClient.java [hunk 2]
```diff
@@ -176,6 +178,7 @@ public String request(String httpVerb, String endpointName, Map<String, Object>
      * @return Unique signature generated using the client_secret and the payload
      */
     public String generateSignature(Map<String, Object> payloadParams, Long timestamp) {
+        logger.info("Generating Signature -");
```
Comment (AsadShakeel) [inline src/main/java/com/veryfi/VeryfiClient.java]: nit: `-` can be removed from logging

File: src/main/java/com/veryfi/VeryfiClient.java [hunk 3]
```diff
@@ -131,7 +131,8 @@ public String request(String httpVerb, String endpointName, Map<String, Object>
      * @return A JSON of the response data.
      * @throws VeryfiClientException if there is any error in making request to veryfi APIs
      */
-    public String request(String httpVerb, String endpointName, Map<String, Object> requestArguments, boolean isFileStream) throws VeryfiClientException {
+    public String processRequest(String httpVerb, String endpointName, Map<String, Object> requestArguments, boolean isFileStream) throws VeryfiClientException {
+        logger.info("Processing Request -[s t a r t]");
```
Comment (AsadShakeel) [inline src/main/java/com/veryfi/VeryfiClient.java]: We can include the method type and endpoint name in the logging for better understanding.
Also, the `httpVerb` can be refactored to `httpMethod`.

File: src/main/java/com/veryfi/VeryfiClient.java [hunk 4]
```diff
@@ -234,13 +239,15 @@ public String getDocument(String documentId) throws VeryfiClientException {
      * @throws VeryfiClientException if there is any error in making request to veryfi APIs
      */
     public String processDocument(String filePath, List<String> categories, boolean deleteAfterProcessing) throws VeryfiClientException {
+        logger.info("Processing Document -");
```
Comment (AsadShakeel) [inline src/main/java/com/veryfi/VeryfiClient.java]: nit: extra `-`

File: src/main/java/com/veryfi/VeryfiClient.java [hunk 5]
```diff
@@ -234,13 +239,15 @@ public String getDocument(String documentId) throws VeryfiClientException {
...
         final String endpointName = "/documents/";
         if (categories == null || categories.isEmpty()) {
             categories = CATEGORIES;
         }
 
         Path fPath = Paths.get(filePath);
         String fileName = fPath.getFileName().toString();
+        logger.info("Process Document -- filePath["+filePath+"],fileName["+fileName+"]");
```
Comment (AsadShakeel) [inline src/main/java/com/veryfi/VeryfiClient.java]: logging formatting can be done here and below all logging statements as well.

**Review by AsadShakeel — CHANGES_REQUESTED** [review]

File: src/main/java/com/veryfi/VeryfiClient.java (hunk 1 above)
Comment (riyazah18) [reply src/main/java/com/veryfi/VeryfiClient.java]: done

File: src/main/java/com/veryfi/VeryfiClient.java (hunk 2 above)
Comment (riyazah18) [reply src/main/java/com/veryfi/VeryfiClient.java]: done

File: src/main/java/com/veryfi/VeryfiClient.java (hunk 3 above)
Comment (riyazah18) [reply src/main/java/com/veryfi/VeryfiClient.java]: done

File: src/main/java/com/veryfi/VeryfiClient.java (hunk 4 above)
Comment (riyazah18) [reply src/main/java/com/veryfi/VeryfiClient.java]: done

File: src/main/java/com/veryfi/VeryfiClient.java [hunk 6]
```diff
@@ -112,26 +112,27 @@ public String getUrl() {
...
+     * @param httpMethod HTTP Method
      * @param endpointName Endpoint name such as 'documents', 'users', etc.
      * @param requestArguments JSON payload to send to Veryfi
      * @param isFileStream A boolean to check if the file stream is available default is false
      * @return A JSON of the response data.
      * @throws VeryfiClientException if there is any error in making request to veryfi APIs
      */
-    public String request(String httpVerb, String endpointName, Map<String, Object> requestArguments, boolean isFileStream) throws VeryfiClientException {
+    public String processRequest(String httpMethod, String endpointName, Map<String, Object> requestArguments, boolean isFileStream) throws VeryfiClientException {
+        logger.info("Processing Request with httpMethod[{httpMethod}],endpointName[{}]-[s t a r t]",httpMethod,endpointName);
```
Comment (AsadShakeel) [inline src/main/java/com/veryfi/VeryfiClient.java]: `httpMethod[{httpMethod}]`  should be replaced by `httpMethod[{}]`
nit: spacing with commas

File: src/main/java/com/veryfi/VeryfiClient.java [hunk 7]
```diff
@@ -234,13 +239,15 @@ public String getDocument(String documentId) throws VeryfiClientException {
...
     public String processDocument(String filePath, List<String> categories, boolean deleteAfterProcessing) throws VeryfiClientException {
+        logger.info("Processing Document");
         final String endpointName = "/documents/";
         if (categories == null || categories.isEmpty()) {
             categories = CATEGORIES;
         }
 
         Path fPath = Paths.get(filePath);
         String fileName = fPath.getFileName().toString();
+        logger.info("Process Document with filePath[{}],fileName[{}]",filePath,fileName);
```
Comment (AsadShakeel) [inline src/main/java/com/veryfi/VeryfiClient.java]: nit: spacing with commas

File: src/main/java/com/veryfi/VeryfiClient.java [hunk 8]
```diff
@@ -270,11 +277,12 @@ public String processDocumentFile(String filePath, List<String> categories, bool
         }
         Path fPath = Paths.get(filePath);
         String fileName = fPath.getFileName().toString();
+        logger.info("Process Document File -- filePath["+filePath+"],fileName["+fileName+"]");
```
Comment (AsadShakeel) [inline src/main/java/com/veryfi/VeryfiClient.java]: nit: logging format and spacing issue

File: src/main/java/com/veryfi/VeryfiClient.java [hunk 9]
```diff
@@ -316,8 +324,9 @@ public String processDocumentUrl(
      * @throws VeryfiClientException if there is any error in making request to veryfi APIs
      */
     public void deleteDocument(int documentId) throws VeryfiClientException {
+        logger.info("Deleting Document -- documentId["+documentId+"]");
```
Comment (AsadShakeel) [inline src/main/java/com/veryfi/VeryfiClient.java]: same
nit: logging format and spacing issue

File: src/main/java/com/veryfi/VeryfiClient.java [hunk 10]
```diff
@@ -328,8 +337,9 @@ public void deleteDocument(int documentId) throws VeryfiClientException {
      * @throws VeryfiClientException if there is any error in making request to veryfi APIs
      */
     public String updateDocument(int id, Map<String, Object> fieldsToUpdate) throws VeryfiClientException {
+        logger.info("Updating Document -- documentId["+id+"]");
```
Comment (AsadShakeel) [inline src/main/java/com/veryfi/VeryfiClient.java]: same
nit: logging format and spacing issue

**Review by AsadShakeel — CHANGES_REQUESTED** [review]

**Review by AsadShakeel — APPROVED** [review]

**Pull request merged**